File: Data Structures & Algorithms/subtree-of-a-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {  
    public boolean isSubtree(TreeNode root, TreeNode sr) {
        if(sr==null) return true;
        if(root==null)return false;

        if(same(root,sr)){
            return  true;
        }
        return isSubtree(root.left,sr) || isSubtree(root.right,sr);
    }
    public boolean same(TreeNode p,TreeNode q){
        if(p==null && q ==null) return true;
        if(p== null || q == null) return false;
        return (p.val==q.val) && same(p.left,q.left)&& same(p.right,q.right);
    }
}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if root is None:
            return []
        qu=deque([root])
        res=[]
        while qu:
            cur=qu.popleft()
            res.append(cur.val)
            if cur.right:
                qu.append(cur.right)
        return res
        